server/src/test/java/org/apache/druid/metadata/SqlSegmentsMetadataManagerTest.java

```diff
@@ -283,7 +283,9 @@ public void testPollPeriodicallyAndOnDemandInterleave() throws Exception
     Assert.assertTrue(sqlSegmentsMetadataManager.getLatestDatabasePoll() instanceof SqlSegmentsMetadataManager.PeriodicDatabasePoll);
     dataSourcesSnapshot = sqlSegmentsMetadataManager.getDataSourcesSnapshot();
     Assert.assertEquals(
-        ImmutableList.of("wikipedia3", "wikipedia", "wikipedia2"),
+    // This test is guava version sensitive when upgrading to guava > 32
+    // the order changes to
+        ImmutableList.of("wikipedia2", "wikipedia3", "wikipedia"),
         dataSourcesSnapshot.getDataSourcesWithAllUsedSegments()
                            .stream()
                            .map(ImmutableDruidDataSource::getName)
```

Comment:
Github won't let me suggest further below in the diff, but let's just remove the ordering assumptions here.

```suggestion
                           .map(ImmutableDruidDataSource::getName).collect(Collectors.toSet())
```